### PR TITLE
[android] #3631 - resize fix issue

### LINKED ISSUE #3631: android:configChanges Activity configuration is broken
It seems that `android:configChanges` is broken (see [this](https://youtu.be/3I5VwwuBy4Q) example)
In our TestApp we have setup the `BulkMarkerActivity` with a `configChanges` configuration:
```xml
<activity
      android:name=".BulkMarkerActivity"
      android:configChanges="orientation|keyboardHidden|screenSize"
      android:label="@string/activity_add_bulk_markers" />
```
I remember that this was working correctly when we added this initially.

### PR DESCRIPTION
**Pull request by tobrun**
closes #3631

### CODE REVIEW

**Comment by tobrun** [comment]
Other related issues are #3622

**Comment by tmpsantos** [comment]
LGTM, :shipit:

**Comment by tobrun** [comment]
@tmpsantos 

Wouldn't it be more clean to use a flag to indicate that we need to call:

```   
if(flagToResize){
 flagToResize = false;
 glViewport(0,0,fbWidth,fbHeight);
}
```

I just tested and it's called a lot when for example panning:

```java
03-08 00:17:52.880 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:52.931 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:52.948 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:52.957 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:52.974 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:52.990 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.007 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.024 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.041 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.058 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.075 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.092 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.109 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.126 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.143 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.159 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.176 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.193 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.227 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.244 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.261 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.278 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.295 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.312 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.329 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.345 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.362 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.379 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.396 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.413 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.430 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.447 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.464 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.498 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.515 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.531 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.548 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.565 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.582 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.599 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.616 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.633 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.650 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.667 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.684 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.701 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.717 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.735 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.751 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.768 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.785 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.802 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.819 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.835 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.852 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.869 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.886 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.903 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.920 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.937 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.954 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.971 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:53.988 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:54.005 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:54.021 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:54.038 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:54.055 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:54.072 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:54.089 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:54.106 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:54.140 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:54.157 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:54.174 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:54.191 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:54.207 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:54.224 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:54.241 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:54.258 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:54.275 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:54.292 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:54.309 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:54.326 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:54.343 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:54.360 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:54.377 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:54.410 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:54.427 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:54.444 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:54.461 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:54.478 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:54.495 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:54.512 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:54.529 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:54.546 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:54.563 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:54.580 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:54.597 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:54.613 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:54.631 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
03-08 00:17:54.647 25074-25093/? D/mbgl: {Map}[Android]: NativeMapView::beforeRender()
```

I'm not sure how to do that correctly in C++ though -_-'

File: platform/android/src/native_map_view.cpp [hunk 1]
```diff
@@ -191,7 +191,10 @@ void NativeMapView::invalidate() {
 
 void NativeMapView::beforeRender() {
     mbgl::Log::Debug(mbgl::Event::Android, "NativeMapView::beforeRender()");
-    // no-op
+    // this fixes the viewport resizing issue,
+    // but we are calling it every time.
+    // we should only call this when the resizeView / resizeFramebuffer is called.
+    glViewport(0,0,fbWidth,fbHeight);
```
Comment (tmpsantos) [inline platform/android/src/native_map_view.cpp]: nit: missing spaces after comma.

`glViewport(0, 0, fbWidth, fbHeight);`

Comment (tobrun) [inline platform/android/src/native_map_view.cpp]: :scream:

**Pull request merged**